packages/calypso-analytics/src/tracks.ts

```diff
@@ -11,8 +13,15 @@ import { loadScript } from '@automattic/load-script';
  */
 import { getCurrentUser, setCurrentUser } from './utils/current-user';
 import getDoNotTrack from './utils/do-not-track';
+import { getPageViewParams } from './page-view-params';
 import debug from './utils/debug';
 
+declare global {
+	interface Window {
+		_tkq: Array< Array< any > >;
+	}
+}
```

Comment:
Nice typescript foo 👍 
hmm this has exposed quite a few places where we weren't checking for `window` to be defined.. I wonder under what circumstances `window` would be undefined...? Some of this code is tested via `client/lib/analytics/test/index.js` and that test code also assumes that `window` exists. I can't think of any reason `window` would be undefined 😂.

Reply:
Some tests run in a `jsdom` environment and some don't, so just because all tests currently assume `window` is available, doesn't mean there won't be tests in the future that don't have `jsdom` but would still like to import this package. Adding `jsdom` to tests does slow them down a little.

Reply:
ahh ok that makes sense, yea this code ideally shouldn't depend on jsdom for testing but it looks like it does for now. I'll just leave those guards around window.